addons/stock/models/stock_production_lot.py

```diff
@@ -65,6 +65,14 @@ def _compute_display_complete(self):
     @api.model_create_multi
     def create(self, vals_list):
         self._check_create()
+        for vals in vals_list:
+            if vals.get('name'):
+                name, product_id, company_id = vals['name'], vals['product_id'], vals['company_id']
+                lot = self.search([('name', '=', name),
+                                   ('product_id', '=', product_id),
+                                   ('company_id', '=', company_id)])
+                if lot:
+                    raise UserError(_('The combination of serial number and product must be unique across a company.\nThe serial number "%s" for product "%s" has already been assigned.' % (lot.name, lot.product_id.name)))
```

Comment:
quant.product_id.display_name, use display_name, if there are variants it is better.

Reply:
You are right. Change applied. Thanks